os/arch/arm/src/armv7-m/up_schedyield.c

```diff
@@ -138,6 +143,11 @@ void up_schedyield(void)
 		/* No, then we will need to perform the user context switch */
 
 		else {
+#ifdef CONFIG_TASK_SCHED_HISTORY
+			/* Save the task name which will be scheduled */
+
+			save_task_scheduling_status(ntcb);
```

Comment:
Why are we not calling this in the` if(current_regs) `condition? Even in that case, a new task is getting scheduled.